packages/uni-virtualizer-examples/public/scroll-to-index-lit-html/index.js

```diff
@@ -0,0 +1,23 @@
+import { render, html } from 'lit-html';
+import { scroll } from 'lit-virtualizer/lib/scroll.js';
+
+const example = (contacts, scrollTo = null) => html`
+    <section style="height: 100%;">
+        ${scroll({
+            items: contacts,
+            template: ({ longText }, i) => html`<p>${i}) ${longText}</p>`,
+            scrollTo: scrollTo,
```

Comment:
Just trying to wrap my head around the API...

As I understand it:

* Application code renders the view, including the scroll directive with a value for `scrollTo`
* If end user then scrolls, logic internal to the scroller stops trying to scroll to the specified index

Is this correct?

If so, what I'm wondering is what happens if / when some later event causes the view to be re-rendered. It seems like the onus is on the application to developer to ensure that the `scrollTo` is cleared before the next render, to avoid passing the now-stale index back into the scroller and triggering an unwanted scroll. Is that right, or am I misunderstanding?

I think this could be OK, but want to make sure we think things through and document intended usage. In particular, I'm wondering what we might advise about how / when to reset `scrollTo` in application state.

Reply:
> Is this correct?

Yup!

> what happens if / when some later event causes the view to be re-rendered

If the developer wants to update their view, they're going to have to call `render` again. The scroll directive takes the whole configuration at once, including `items` and `template`. So in my mind this is consistent, in that they must consider the whole configuration whenever they deliberately render.

The `<lit-virtualizer>` _element_ will be affected by your concern I think. For `<lit-virtualizer>`, the developer can set the `.items` and `.template` properties separately, and will be able to call `scrollToIndex` discretely as well. In this scenario, I agree, they shouldn't have to worry about a stale index. Luckily we can use private internal state in `LitVirtualizer` to make sure `scrollToIndex` doesn't cause a permanent scroll state. (commit incoming for the `<lit-virtualizer>` support)